### DISCUSSION TITLE
ImageMagick Policy.xml, what if I comment out the memory policy domain line

### DISCUSSION THREAD

**Discussion by jetpack1116** [ImageMagick/ImageMagick#4390] [Help]
In ImageMagick Policy.xml, what if I comment out the memory policy domain line, what will happen.
I observed that is uses a default maximum limit?

Is this the maximum known limit for IM of a specific version?
Since if I uncomment the policy domain memory setting entry and set a value greater that the maximum limit, it cannot be set.

**Comment by snibgo** [comment]
See https://www.imagemagick.org/script/architecture.php

If you have no entry for "memory", then IM won't set a limit on memory usage for pixel caches.

I don't know what you mean by "maximum limit". It is pointless trying to set a limit that is greater than the amount of physical memory in your computer.

**Reply by jetpack1116** [reply]
@snibgo but if you execute the command `identify -list resource` , a memory limit is shown

**Reply by jetpack1116** [reply]
Are there any documentation regarding that? Because if that is the case we always need to calculate memory usage inorder to optimize...

**Reply by jetpack1116** [reply]
When I execute `identify -list resource` for IMver 7.0.8-6, the memory limit shown is 3Gb
When I execute `identify -list resource` for IMver 7.1.0-0, the memory limit shown is 8Gb

This is in same PC and the memory entry in policy.xml is removed/commented out.

Is this a bug of ImageMagick?

**Comment by snibgo** [comment]
If no entry for memory is given in policy.xml and there is no environment variable MAGICK_MEMORY_LIMIT then IM calculates a default value, in resource.c ResourceComponentGenesis(). Perhaps the code has changed between IM versions.

**Comment by dlemstra** [comment]
That calculation for the default memory limit has changed between versions.

**Reply by jetpack1116** [reply]
@dlemstra Thanks. So you mean for each version of ImageMagick there is a default memory limit?

**Reply by dlemstra** [reply]
Not sure what you are asking. But we probably made a change/fix in the calculation between those two versions.
